.github/workflows/test.yaml

```diff
@@ -31,4 +35,13 @@ jobs:
         if: always()
```

Comment:
Can't this be removed?

Reply:
No, it makes sure the artifact is always uploaded, even if previous steps fail. The previous test step is pretty long and we want the coverage results to be always uploaded even when the entire step might fail, otherwise we will not be able to debug.